resource.go

```diff
@@ -59,7 +61,14 @@ func LoadResourceFromPath(path string) (Resource, error) {
 
 // LoadResourceFromURLString creates a new StaticResource in memory using the body of the specified URL.
 func LoadResourceFromURLString(urlStr string) (Resource, error) {
-	res, err := http.Get(urlStr)
+	req, err := http.NewRequest(http.MethodGet, urlStr, nil)
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
timeout ok?

Reply:
I have to wonder what other things would benefit from adding context, and if the timeout should be stored somewhere where it can be used across these places?

Reply:
There probably are more places, yes. If you can think of any, feel free to point me there.

I changed improved the code so users can also set the context themselves.

Reply:
It might be simpler to just add `Timeout` to client - does the context add more functionality than the timeout?
```
client := http.Client{
    Timeout: time.Second * 60,
}
res, err := client.Get(url)
```

Reply:
It is equivalent if we don't add the 'LoadResourceFromURLStringWithCtx' function.

If we do, the context is superior as it does not only allow timeouts but can also e.g. be cancelled early.